### PR TITLE
Convert our current typing to implicit typing

### PR DESCRIPTION
**Pull request by vindard**
## Description

The goal of this PR is to remove `import` statements for types and to have them available automatically in the environment using Typescript's global type declarations functionality (see #347)

The structure we decided on was to have `.d.ts` files as close as possible to the module they refer to, and to have a global `types.d.ts` file for types that don't map cleanly to singular modules. The aim would be to minimize what goes into this `types.d.ts` file and if possible eventually remove that file.

### Outstanding `TODO`:
- [x] _(Done https://github.com/GaloyMoney/galoy/pull/382/commits/0610231155630bebef2fabfdb6b25414b61da84f)_ ~Split types from `types.d.ts` into separate files closer to relevant modules where possible~

### CODE REVIEW

**Comment by vindard** [comment]
@samerbuna it seems that the eslint error might be [a legitimate error](https://github.com/typescript-eslint/typescript-eslint/issues/955#issuecomment-544223931)?

For example if I try to create `src/Lightning.d.ts` alongside `src/Lightning.ts` I get:
```
Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: src/Lightning.d.ts.
The file must be included in at least one of the projects provided.
```

File: src/types.d.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
+type Logger = import("pino").Logger
+type UserType = typeof import("./schema").User
```
[the remark this answers is not in the document]
Comment (samerbuna) [reply src/types.d.ts:2]: I like the `*Type` convention but I I think namespaces are a better way to go. Also keep an eye on [the new schema](https://github.com/GaloyMoney/galoy/pull/332) to see the new names coming down the pipeline.

[the remark this answers is not in the document]
Comment (vindard) [reply src/types.d.ts]: Ok, should I change these variable renames as well then? -> https://github.com/GaloyMoney/galoy/pull/382/commits/55b38dbb38a0db163c4158430f19db998bbd3089

Specifically I did:
- `Level` -> `level`/`Level`
- `OnboardingEarn` -> `onboardingEarn`

**Comment by vindard** [comment]
I did all the splitting for the global types file into separate module-level declaration files in this commit -> https://github.com/GaloyMoney/galoy/pull/382/commits/0610231155630bebef2fabfdb6b25414b61da84f

**Review by bodymindarts — APPROVED** [review]
LGTM!

**Review by samerbuna — APPROVED** [review]

**Pull request merged**